src/MeshInertiaCalculator.cc

```diff
@@ -205,44 +203,42 @@ std::optional<gz::math::Inertiald> MeshInertiaCalculator::operator()
     return std::nullopt;
   }
 
-  auto fullPath = asFullPath(sdfMesh->Uri(), sdfMesh->FilePath());
+  const common::Mesh *mesh = loadMesh(*sdfMesh);
 
-  if (fullPath.empty())
-  {
-    gzerr << "Mesh geometry missing uri" << std::endl;
-    _errors.push_back({sdf::ErrorCode::URI_INVALID,
-        "Attempting to load the mesh but the URI seems to be incorrect"});
-    return std::nullopt;
-  }
-
-  // Load the Mesh
-  gz::common::MeshManager *meshManager = gz::common::MeshManager::Instance();
-  mesh = meshManager->Load(fullPath);
-  if (!mesh)
+  // Compute inertia for each submesh then sum up to get the final inertia
+  // values.
+  gz::math::Inertiald meshInertial;
+  for (unsigned int i = 0; i < mesh->SubMeshCount(); ++i)
   {
-    gzerr << "Failed to load mesh: " << fullPath << std::endl;
-    return std::nullopt;
-  }
-  std::vector<Triangle> meshTriangles;
-  gz::math::MassMatrix3d meshMassMatrix;
-  gz::math::Pose3d centreOfMass;
+    std::vector<Triangle> meshTriangles;
+    gz::math::MassMatrix3d meshMassMatrix;
+    gz::math::Pose3d centreOfMass;
 
-  // Create a list of Triangle objects from the mesh vertices and indices
-  this->GetMeshTriangles(meshTriangles, sdfMesh->Scale(), mesh);
+    // Create a list of Triangle objects from the mesh vertices and indices
+    auto submesh = mesh->SubMeshByIndex(i).lock();
+    this->GetMeshTriangles(meshTriangles, sdfMesh->Scale(), submesh.get());
 
-  // Calculate mesh mass properties
-  this->CalculateMassProperties(meshTriangles, density,
-    meshMassMatrix, centreOfMass);
+    // Calculate mesh mass properties
+    this->CalculateMassProperties(meshTriangles, density,
+      meshMassMatrix, centreOfMass);
 
-  gz::math::Inertiald meshInertial;
+    if (meshMassMatrix.IsValid())
+      meshInertial += gz::math::Inertiald(meshMassMatrix, centreOfMass);
+  }
 
-  if (!meshInertial.SetMassMatrix(meshMassMatrix))
+  if (meshInertial.MassMatrix().Mass() > 0.0 &&
+      meshInertial.MassMatrix().IsValid())
   {
-    return std::nullopt;
+    return meshInertial;
   }
   else
   {
-    meshInertial.SetPose(centreOfMass);
-    return meshInertial;
+    gzerr << "Failed to computed valid inertia in MeshInertiaCalculator. "
```

Comment:
super nit: I think we can remove the `else` case for readability
```suggestion
if (!meshInertial.MassMatrix().IsValid()) {
  gzerr << "Failed to computed valid inertia in MeshInertiaCalculator. "
  ...
  _errors.push_back({sdf::ErrorCode::WARNING,
        "Could not calculate valid mesh inertia"});
  return std::nullopt;
}

return meshInertial;
```

Reply:
removed else in [b414d0e](https://github.com/gazebosim/gz-sim/pull/2770/commits/b414d0eeca77a5a626804129f27a3b74db568747)